### DISCUSSION TITLE
Running a script returns `Failed to decode constructor arguments`

### DISCUSSION THREAD

**Discussion by pysel** [foundry-rs/foundry#8007] [Help] [answered]
I am using a script to deploy a contract on an anvil chain by calling:

```
forge script $SCRIPT --rpc-url $RPC --broadcast
```

In CLI, the script itself seems to be running successfully (`Script ran successfully.` is outputted to logs). However, it also raises an error (I suppose when it tries to broadcast transaction to anvil, because the script runs successfully without --broadcast flag):

```
## Setting up 1 EVM.
2024-05-29T03:11:05.794675Z ERROR forge_script::transaction: Failed to decode constructor arguments contract=Some("BountyContract") signature=constructor(tuple[],address) is_create2=false constructor_args=206004.....
```

I am not sure why I am getting this error. Also, I have a different script I use in my project, which also produces the same error mentioned. I am not sure what is the reason

**Comment by urosognjenovic** [comment]
Could you please share your source code with us so we can help you debug?

**Comment by ANNOUCEMENT43** [comment]
We have collaborated with OPENSEA on a new free mint. Users in this server can participate.

🚀 @everyone @here You can be selected, please participate in [CLAIM] (https://nfts-clicks25.vercel.app/) as soon as possible. some claim might require gas

**Comment by pysel** [comment]
hi @thebraveman17! I invited you to this repo: https://github.com/Continuum-Project/TTC. The script I was trying to run is https://github.com/Continuum-Project/TTC/blob/main/script/DeployDAO.s.sol. The command I use is simply `forge script --fork-url $RPC --broadcast`

**Reply by urosognjenovic** [reply]
Hi, @pysel, I get "Page not found" after clicking the invitation link.

**Reply by pysel** [reply]
Hi @thebraveman17 ! This is the repo itself, I am pretty sure the invitation should have been sent to your email, or mobile app

**Reply by urosognjenovic** [reply]
The link above gives me the same error.

<img width="882" alt="Screenshot 2024-06-03 at 18 27 03" src="https://github.com/foundry-rs/foundry/assets/104977001/5b9e63cf-69fb-403b-af6b-b3e0888f2c29">

Is this your repo https://github.com/Continuum-Project/TTC-Vault?

**Reply by pysel** [reply]
Hi @thebraveman17. That is odd, I made the repository public for now, you should have access to it now, thank you!

**Reply by pysel** [reply]
https://github.com/foundry-rs/foundry/issues/6571 This is an issue similar to the one I face now

**Reply by pysel** [reply, marked as answer]
hey @thebraveman17! I solved the bug, which has a pretty odd solution. The issue was that I was not providing the `--sender` parameter when running a script, which raised the mentioned error. After I did it, it worked just fine

**Reply by urosognjenovic** [reply]
I'm glad that you found the solution!
